### DISCUSSION TITLE
Cannot boot from local disk (stuck on "Booting from SAN device...")

### DISCUSSION THREAD

**Discussion by Vinjul1704** [ipxe/ipxe#740] [Q&A]
Hello, I am trying to boot from a local disk, however trying to do that via `sanboot --no-describe --drive 0x81` command results in it being stuck in iPXE, with the cursor/underscore blinking on the line underneath.

I tried booting Windows 10 installed on an SSD and Ventoy installed on a flash drive like that, both resulting in the same issue. iPXE is run off a flash drive and the PC is a simple Kaby Lake consumer board with a Pentium G4560 and no additional network hardware installed (using an Intel Pro/10000 PT results in the same issue).

Any ideas are appreciated, thank you!

**Comment by NiKiZe** [comment]
What are you trying to boot?
If windows I do t think it supports booting on secondary drive unless it was installed on it. But you might be able to modify the BCD to get it working.

(I'm assuming this is pcbios, not efi?)

**Reply by Vinjul1704** [reply]
I am trying to boot a Windows 10 installation that's already installed on an SSD attached to the motherboards SATA ports. It should be efi, but I couldn't find a CSM toggle in the bios (the board in question is the "Gigabyte GA-B250M-D2V", in case that matters).

**Reply by Vinjul1704** [reply]
I also just tried it on a different PC, this time on an AM4 platform ("ASRock X570M Pro4" to be exact), and it has the same issue. CSM had to be enabled in order for the iPXE flash drive to show up in the boot list.

**Reply by NiKiZe** [reply]
If you want to boot in efi mode you need a efi build of ipxe, by default windows can only boot on the platform it was installed on.

**Reply by Vinjul1704** [reply]
How would I go about building it as an ISO I can boot from a flash drive? I only found information about that it "should be merged" and about building EFI ROMs to flash on to network adapters.

Also, forgot to mention it, but on the AM4 platform, I was trying to boot Arch Linux instead of Windows, which has GRUB installed for both EFI and legacy BIOS, but the issue stayed the same.

**Comment by Vinjul1704** [comment]
I just got around to do a little more testing and decided to boot the disk via the network instead, which results in the same issue. 

The disk in question has a working Windows 10 installation on it, is connected to a PC running Linux, which serves it as a target on the network. I am able to use "sanhook" to connect to it, but booting from it results in the same issue as before, so it getting stuck on "Booting from SAN device 0x80".

I tried it using iPXE compiled for both pcbios and efi, in both cases booted on the server via a flash drive.

Any help or ideas would be appreciated!

**Reply by NiKiZe** [reply]
A Windows install by default only ha pcbios or efi boot. You can not mix this up.

**Reply by Vinjul1704** [reply]
> A Windows install by default only ha pcbios or efi boot. You can not mix this up.

What do you recommend I try next? As I said, I tried it using both a pcbios as well as an efi build of iPXE.

**Reply by NiKiZe** [reply]
I suggest you verify what you have. Guessing is not a winning concept. 😉
Also, even if you do get the disk to actually boot it will probably bluscreen right away.

Have you seen the documentation on ipxe.org? Or checked other discussions here on iSCSI?

**Reply by Vinjul1704** [reply]
I have checked the documentation and discussions on here before, but finally got it working after looking through some more things. In particular, this is what I changed this time (not sure which one was the issue now):

- Enabled framebuffer console instead of using whatever the default one is and set it to my monitor resolution (1920x1200).
- Figured out why there was no CSM toggle in my BIOS, made it visible and disabled CSM.

This then allowed me to actually boot the SSD over the network, but it blue-screened during boot as you predicted. Since my end goal is to boot that disk locally anyway and have a second disk be connected over the network before that, I decided to give this MR/branch a try: https://github.com/ipxe/ipxe/pull/612

This ended up working perfectly and now my setup is where it should be. Out of curiosity, is there any reason why that MR hasn't been merged yet?
